R/mod_demographics_selection.R

```diff
@@ -0,0 +1,107 @@
+#' demographics_selection UI Function
+#'
+#' @description A shiny Module.
+#'
+#' @param id,input,output,session Internal parameters for {shiny}.
+#'
+#' @noRd
+#'
+#' @importFrom shiny NS tagList
+mod_demographics_selection_ui <- function(id) {
+  ns <- NS(id)
+  tagList(
+    uiOutput(ns("dynamic_demographics_selection"))
+  )
+}
+
+#' demographics_selection Server Functions
+#'
+#' @noRd
+mod_demographics_selection_server <- function(id, filter_data) {
+  moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+
+    # the UI render
+
+    output$dynamic_demographics_selection <- renderUI({
+      # check which demographic variables are present
+
+      isolate({
+        has_age <- "age" %in% colnames(filter_data()$unique_data)
+        has_gender <- "gender" %in% colnames(filter_data()$unique_data)
+        has_ethnicity <- "ethnicity" %in% colnames(filter_data()$unique_data)
+      })
+
+      tagList(
+        if (has_age) {
+          uiOutput(ns("age_UI"))
+        },
+        if (has_gender) {
+          uiOutput(ns("gender_UI"))
+        },
+        if (has_ethnicity) {
+          uiOutput(ns("ethnicity_UI"))
+        }
+      )
+    })
+
+    # demography selection----
+
+    output$age_UI <- renderUI({
+      isolate(
+        choices <- filter_data()$unique_data %>%
+          dplyr::arrange(age) %>%
+          dplyr::distinct(age, .keep_all = TRUE) %>%
+          dplyr::filter(!is.na(age))
+      )
+
+      choices <- factor(choices$age,
```

Comment:
It's a minor point but overwriting a dataframe (`choices`) with a vector seems a bit counter intuitive. Maybe better to have `choices <- factor(choices_df$age...)`?